Update packagename new spilt solution

Amichelangelo opened the pull request:
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

<a href="?expand=1&template=sdk_configuration_template.md">Click here</a> to open a PR for only SDK configuration.

Qi-2077 committed:
1. rename import to begin_import_assets avoid breaking changes
2. QuestionAnsweringProjectMetadata -> QuestionAnsweringProject (SDK review)
3. getSynonyms -> listSynonyms (SDK review)

Qi-2077 committed:
rename

Qi-2077 committed:
update

Qi-2077 committed:
update Azure.AI.Language.QuestionAnswering to Azure.AI.Language.QuestionAnswering.Interface

Qi-2077 committed:
update-questionAnwseringAuthoring-sdkReview

Qi-2077 committed:
update authoring

sarkar-rajarshi commented on specification/cognitiveservices/data-plane/LanguageQuestionAnsweringAuthoring/routes.tsp:
```diff
@@ -19,26 +19,78 @@ alias ServiceTraits = NoRepeatableRequests &
+}
+
+/** Parameters for the import project operation. */
+model ImportJobParameters {
```
Same here

sarkar-rajarshi commented on specification/cognitiveservices/data-plane/LanguageQuestionAnsweringAuthoring/models.tsp:
```diff
@@ -347,7 +347,7 @@ model UpdateSourceFiles {
  */
 @resource("authoring/query-knowledgebases/projects")
-model QuestionAnsweringProjectMetadata {
+model QuestionAnsweringProject {
```
Why rename?

Amichelangelo replied:
<img width="1379" height="250" alt="image" src="https://github.com/user-attachments/assets/634fce35-12e9-4404-ba5c-d738840a8b52" />
<img width="1687" height="262" alt="image" src="https://github.com/user-attachments/assets/ec7b1ecc-0222-47a1-ac73-36f99e9f6845" />
due to the comments from SDK review

sarkar-rajarshi commented on specification/cognitiveservices/data-plane/LanguageQuestionAnswering/client.tsp:
```diff
@@ -3,10 +3,10 @@ import "@azure-tools/typespec-client-generator-core";
 
 @useDependency(Versions.v2025_05_15_preview)
-namespace Language.QuestionAnsweringClientCustomizations;
+namespace Language.QuestionAnswering.InferenceClientCustomizations;
```
Do you need to change all of these?

sarkar-rajarshi commented on specification/cognitiveservices/data-plane/LanguageQuestionAnswering/client.tsp:
```diff
@@ -33,10 +33,10 @@ namespace Language.QuestionAnsweringClientCustomizations;
 @client({
   name: "QuestionAnsweringClient",
-  service: Language.QuestionAnswering,
+  service: Language.QuestionAnswering.Inference,
```
I am curious to see how these values are affecting code generation? If they don't affect code generation, I'd not change them

Amichelangelo replied:
remove this change

sarkar-rajarshi reviewed (commented):
left a few comments

Qi-2077 committed:
update

Amichelangelo commented on specification/cognitiveservices/data-plane/LanguageQuestionAnsweringAuthoring/routes.tsp:
```diff
@@ -19,26 +19,78 @@ alias ServiceTraits = NoRepeatableRequests &
 alias LanguageOperations = ResourceOperations<ServiceTraits, ErrorResponse>;
 
+/** Parameters for the export project operation. */
+model ExportJobParameters {
```
removed

Qi-2077 committed:
revert uncorrect changes

Qi-2077 committed:
output-dir

Qi-2077 committed:
remove interface

sarkar-rajarshi reviewed (approved):
LGTM

Qi-2077 committed:
Merge branch 'main' into qiyin/update-packagename-new-spilt-solution

Qi-2077 committed:
remove old emit opiton and update namespace and output dir

Qi-2077 committed:
test

Qi-2077 committed:
test

Qi-2077 committed:
correct formart

Qi-2077 committed:
test

Qi-2077 committed:
update

Amichelangelo committed:
update

Qi-2077 committed:
fix comments, get-> list

Qi-2077 committed:
add rename for QuestionAnsweringProjectMetaData

Qi-2077 committed:
extract model, rename format

Amichelangelo committed:
Renamed ImportOptions → ImportQueryParameters
Renamed ExportOptions → ExportQueryParameters
Kept ImportJobOptions body outside ImportQueryParameters definition
Applied @@clientName for Python to rename format (conflicts with built-in function)

Qi-2077 committed:
tsp format

sarkar-rajarshi commented on specification/cognitiveservices/data-plane/LanguageQuestionAnsweringAuthoring/routes.tsp:
```diff
@@ -13,6 +13,33 @@ using Azure.Core.Traits;
+/**
+ * Optional query parameters for the import operation.
+ */
+model ImportQueryParameters {
```
move inside models.tsp

sarkar-rajarshi reviewed (approved):
move the two new objects inside models.tsp. rest is good

Qi-2077 committed:
move ExportQueryParameters and ImportQueryParameters inside models.tsp

sarkar-rajarshi committed:
Merge branch 'main' into qiyin/update-packagename-new-spilt-solution
